openedx/core/djangoapps/content/course_structures/migrations/0001_initial.py

```diff
@@ -0,0 +1,46 @@
+# -*- coding: utf-8 -*-
+import datetime
+from south.db import db
+from south.v2 import SchemaMigration
+from django.db import models
+
+
+class Migration(SchemaMigration):
+
+    def forwards(self, orm):
+        # Adding model 'CourseStructure'
+        db.create_table('course_structures_coursestructure', (
+            ('id', self.gf('django.db.models.fields.AutoField')(primary_key=True)),
+            ('created', self.gf('model_utils.fields.AutoCreatedField')(default=datetime.datetime.now)),
+            ('modified', self.gf('model_utils.fields.AutoLastModifiedField')(default=datetime.datetime.now)),
+            ('course_id', self.gf('xmodule_django.models.CourseKeyField')(max_length=255, db_index=True)),
+            ('version', self.gf('django.db.models.fields.CharField')(max_length=255, null=True, blank=True)),
+            ('structure_json', self.gf('django.db.models.fields.TextField')()),
+        ))
+        db.send_create_signal('course_structures', ['CourseStructure'])
+
+        # Adding unique constraint on 'CourseStructure', fields ['course_id', 'version']
+        db.create_unique('course_structures_coursestructure', ['course_id', 'version'])
```

Comment:
As a very random note, we've run into problems before because MySQL was persisting datetime fields without fractional seconds, so choosing the most recent thing based on a datetime field could yield mixed results if there were multiple updates within a second. To fix this, we used the `id` as a tie-breaker in the ordering, since the most recently added things would come later. I don't know if this is still a problem, but it's something to be aware of.